airbyte-integrations/connectors/source-mssql/src/main/java/io/airbyte/integrations/source/mssql/MssqlSource.java

```diff
@@ -96,6 +104,40 @@ public MssqlSource() {
     super(DRIVER_CLASS, AdaptiveStreamingQueryConfig::new, new MssqlSourceOperations());
   }
 
+  @Override
+  public ConnectorSpecification spec() throws Exception {
+    if (cloudDeploymentMode()) {
+      final ConnectorSpecification cloudDeploymentSpec = Jsons.clone(super.spec());
+      // Remove "unencrypted" value for "ssl_method".
+      ((ArrayNode) cloudDeploymentSpec.getConnectionSpecification().get("properties").get(SSL_METHOD).get("oneOf")).remove(0);
+      return cloudDeploymentSpec;
+    }
+    return super.spec();
+  }
+
+  @Override
+  public AirbyteConnectionStatus check(final JsonNode config) throws Exception {
+    // #15808 Disallow connecting to db with disable, prefer or allow SSL mode when connecting directly
+    // and not over SSH tunnel
+    if (cloudDeploymentMode()) {
+      if (config.has(TUNNEL_METHOD)
+          && config.get(TUNNEL_METHOD).has(TUNNEL_METHOD)
+          && config.get(TUNNEL_METHOD).get(TUNNEL_METHOD).asText().equals(NO_TUNNEL)) {
+        // If no SSH tunnel.
+        if (config.has(SSL_METHOD) && config.get(SSL_METHOD).has(SSL_METHOD) &&
+            SSL_METHOD_UNENCRYPTED.equalsIgnoreCase(config.get(SSL_METHOD).get(SSL_METHOD).asText())) {
```

Comment:
I'd also consult with @stephane-airbyte whether there are new unencrypted modes he's adding as part of his job of refitting ssl on this connector - a "prefer ssl" mode for example.

Reply:
nah, the ms-sql JDBC protocl is widely different than the postgres or mysql. You can push as is